Cargo mail delivery (#405)
space-wizards/docs: src/en/space-station-14/departments/cargo-salvage/proposals/mail-delivery.md

# Cargo Mail Delivery

| Designers | Implemented | GitHub Links |
|---|---|---|
| ScarKy0 | :x: No | TBD |

## Overview

This design doc is meant to introduce the concept of mail deliveries to the cargo gameplay loop. Allowing cargo to deliver mail to specific people on the station in order to gain some bonus money.

## Background

Currently cargo's main way of earning money is the bounty system and selling random objects around the station. This can lead to situations such as other departments not cooperating or some bounties simply being not possible to make within a reasonable timeframe.
While those issues can be fixed by making changes to the bounty system (like lowering the bounty skip cooldown) that feels like a bandaid fix to the fact cargo has basically only one way to earn money.

> Delivering mail is not meant to replace bounties, simply work as a small side thing that can be done in case all bounties are currently being stalled.

Reviewer: I would not put the focus on it being a backup way to earn spesos, as this implies it should not be used when bounties are being handled. Instead it has its own merits, even if less profitable.
- social interactions between players & department interaction
- free stuff for the player receiving mail
- more gameplay variety

Reviewer 2: Mail in other forks is historically a really introductory role/task for Cargo as well, moreso than Tech. Gives players the chance to see how Cargo works on a personal level without being given a role that can extremely stressful in a pinch.

## Features to be added

The main things needed are the mail teleporter, as well as letters and packages.

Letters and packages spawn inside of the mail teleporter, which can be opened by anyone with cargo access to drop the currently held items.

> Said items then have to be delivered to a randomly picked crewmember on the station (protected by some filter, so Operator Gamma doesn't suddenly receive mail). On delivery the desired crewmember opens it using their ID and cargo gets a small amount of money.

Reviewer: The id mechanics need some more details, especially for the technical implementation. Is anyone on the crew manifest picked as a target? How is this connected to their ID? How do we prevent cargo from using an ID card console and a bunch of blank IDs to fake the recipient by changing them to their name? Should agent IDs be able to bypass this?

If a package is delivered, on opening it should also grant the target some silly item or food for sure, maybe very rarely minor contraband or things like a pipebomb.

> Possible means of this to be expanded could include a rigged package that could work as a cargo syndie-only item, as well as an ability to make custom packages as a more discreet way of transporting items secretly to others.

Reviewer 2: A nice implementation on the letters could be something themed to the item as well or unique little messages. "Trial of our new product" spam messages with a delicious soda, "Send this letter to five others or YOU CURSED" and such. Nothing too personal like a letter from family, but fun kinds of space nonsense.

For Syndies specifically rather than a rigged package as a singular static item, it might be better to give them an option to stealthily open mail and slip in whatever they want to tie it directly into the second ability idea. This would give Syndie cargo players a lot more freedom with how they approach the idea while also being a simple interaction to implement (And maybe the Detective gets the ability to see tampered mail or something along those lines). For LRP it'd more than likely just be used to remote deliver explosives, but on higher RP forks it could easily be used to tamper with letters to deliver more personal stories or threats, or coercion!

The general idea for cargo timers seems to be roughly 15 minutes which might be a good implementation of when mail arrives to be delivered.

Main things to consider:

> - Delivery items need to have a timer that starts upon being taken out of the teleporter. When the timer runs out the money earned should be significatly lowered, either gradually or instantly. This encourages cargo to actually deliver the mail instead of leaving it at the front for others to check.

Reviewer: The teleporter needs more details as well. Where is it placed, on the ATS or in cargo? How often does it spawn new mail? Only when existing ones are delivered, on a schedule, or are there events affecting this?

> - A delivery opened by a non-recipient should bring some sort of downside. Possibly taking away money from cargo so they cannot use this as a free item generator of sorts.

Reviewer: Should a non-reipient be able to open the mail at all? Or should it be restricted to the ID owner?

## Game Design Rationale

Currently cargo is oriented around the bounty system (except salvage, who is mostly seperate from the main cargo loop). A mail system allows for more interaction between players as well as lowers dependancy on bounties.

Cargo is already meant to interact with other departments, this plays into that fact and gives them a little something to do in the meantime. Ideal situation could be delivering a package to a botanist at the same time as you want to grab a bounty from them. Or a syndicate cargo tech delivering a fake package with a comically timed pimebomb to explode when opened.

## Roundflow & Player interaction

> At the start of the round, a cargo technician visits the ATS (or wherever the mail teleporter will be mapped) and grabs letters or a package to deliver. Upon checking the manifest they may also pick some bounties to deliver to that person at the same time. Upon delivery cargo gets a few spare spesos to spend on bonus steel or similiar while the recipient maybe gets a bike horn to annoy security with.

Reviewer 2: How will deliveries be tracked? Will it reward the spesos once the letter is opened, or is it specifically on delivery? If it's specifically on delivery: what will be used to track that for Cargo so they can see the progress of letters, or even if it went to the right person?

If it's on opening: What happens if the techie delivers a letter and the person just never opens it, or throws it away? Technically delivered, but not mechanically.

> Interactions with other players are key here, both sides benefit slightly as well as get to interact more with each other. Cargo should not simply leave mail at their front door and get someone to pick it up whenever they walk past. The penalty timer is meant to have them deliver it instead. Leaving the packages at front ready for pickup also means a random passerby may open all of them and lead to big penalties, further discouraging this method of delivering mail.

Reviewer 2: Also: Where will the letters be stored and held? If one person wants to handle mail specifically (If Cargo is overcrowded for instance), should they have a handheld satchel for mail delivery, or should they be dragging a large crate around?

Additionally, mail is not meant to be a free item generator for cargo to use to flex drip they get. Losing money for opening mail that doesn't belong to them means they it is almost never worth it to open a package while not the recipient, other than possibly sabotaging cargo if they slip up.

# Technical Considerations

> Mainly a new DeliveryComponent, DeliverySpawnerComponent and a DeliverySystem to manage spawning and interacting with the packages. Impact on performance is most likely going to be minimal.

Reviewer: You just came up with a few names, not with any technical considerations. See the comments above.

This is meant to serve several things:
- Allow new players to get accustomed to learning cargo easier. Currently a new cargo tech might have no idea how to complete bounties or earn money properly. A simple task of delivering mail can help teach the station layout and introduce to the game.
- Add gameplay variety to cargo. Cargo as it stands has a rather simple gameplay loop. You print a bounty, you deliver the invoice to a department and then you sit at cargo until you're told it is ready for pickup. Not only is this repetitive but several bounties can also be completed on the spot, which makes cargo generally boring gameplay wise.
- Encourage player interaction. Simple mail deliveries can greatly help with interactions between cargo and other players/departments.
- Add a little bit of chaos to the station. Mail ultimately can grant some silly random items to the recipient, and what is more fun than finding a bike horn or some bananas to annoy others with?

## Features to be added

### Mail Teleporter

The Mail Teleporter is the main thing controlling the flow of mail to the station. It should be mapped somewhere within cargo (being station or ATS).

> While connected to power and enabled, it will generate mail on top of itself every few minutes up. Anyone with cargo access can toggle it on and off to prevent further mail from spawning.

Author: I have no idea how destruction of it can be handled, cargo sell consoles can also simply be destroyed so for now i'll lean into the QM having a spare board.

Also for a constant flow of mail i am not quite sure, as it can indeed just turn into wordless throwing mail at people.

### Deliveries

Deliveries refer to anything that can spawn using the Mail Teleporter. The main focus here is Letters and Packages.
On spawn, deliveries get assigned a random player in the crew manifest. When collected by the desired recipient they must scan their fingerprint on the delivery to open it, granting them a little reward and cargo some bonus money.
If a non-recipient is to try to open a delivery, it will instead beep and say they are unable to open it. This however can be bypassed by tearing the delivery open (for letters) or smashing it open (for packages). Doing so will invoke a penalty on cargo, taking away from their funds, but will still grant the player whatever was inside.

#### Letters

> A small item, usually containing spam mail, birthday wishes (maybe even with money!) and similiar.

Reviewer 3: Money is only relevant to cargo right now.

Author: Doesnt mean it always will in the future, its just a small flavor to add to the letters

They can be torn open using a Verb if the current holder is unable to open them. Obviously giving cargo a penalty (as stated above) but granting the user the contents.

#### Packages
A big item that occupies both hands. They are supposed to contain things slightly more of value or items that can be useful to your average player. (Tools, Cakes, Plushies, rarely minor contraband and pipebombs)
When on the ground it can be smashed open with any weapon to break the seal, allowing access to whatever is stored inside, still granting cargo a penalty.

> ### Future Expansions

Reviewer 3: Have this tie into future science content. Bluespace mail bag that has an expanded capacity for mail? Mail China lake launcher that automatically places items (the package) in a user's hand when they're hit? Etc.

#### Syndie Uplink Items

Deliveries don't have to stop at just letters and packages. There is also the possibility of a cargo-only uplink item that is a rigged package, potentially spawning something hostile or exploding.

> There could also be some way to sneak items into packages, having the recipient leave their sweet fingerprints on the contents and allowing for a way to frame them.

Reviewer 3: The ability to address a package to someone in-round would have to be considered if a Syndicate agent was to ship their target a pipe bomb.

You could remark that the mix of usually good items would offset the meta-tendency to reject the delivery of a package due to it being a potential pipe bomb.

Although, considering that this promotes player interaction and most syndicates are just looking for their target to do something, the mail system could get them close enough.

#### Custom Mail

> Ever wanted to give your arch-nemesis a threatening letter, but never wanted to give it in person? There is potential for a way to create custom letters/packages and allow cargo to deliver them for you.

Author: Again, the "Future Expansions" just is meant to show the mail system has potential to grow and isn't meant as an actual direction for the future.

- Items granted by deliveries cannot hold big value to them, as to not make breaking packages open and selling their contents more profitable than delivering them.
- Cargo should have a reason to not allow their deliveries to be stolen, as that obviously not only makes them lose out of money, but also gives them a penalty if the seal is broken by a 3rd party.

## Game Design Rationale

Currently cargo is oriented around the bounty system (except salvage, who is mostly seperate from the main cargo loop). A mail system allows for more interaction between players as well as lowers dependancy on bounties.

Cargo is already meant to interact with other departments, this plays into that fact and gives them a little something to do in the meantime. Ideal situation could be delivering a package to a botanist at the same time as you want to grab a bounty from them. Or a new player player grabbing a bunch of deliveries and getting to explore the station and meet other players as they deliver them.

> ## Roundflow & Player interaction

Author: I believe someone is making a PR that allows for stat tracking for end of round stuff like this, but ultimately it doesn't really concern the design doc

At the start of the round, a cargo technician visits the ATS (or wherever the mail teleporter will be mapped) and grabs letters into their bag and a few packages. Upon checking the manifest they may also pick some bounties to deliver to that person at the same time. The cargo tech brings the mail to the recipient, who then unlocks it and grants cargo some bonus money and gets something small in return.
In the case the delivery is simply never opened, it will slowly lose all value and become free game for whoever finds it and decides to open it.

Interactions with other players are key here, both sides benefit while being able to get to interact more with each other. Cargo should not simply leave mail at their front door and get someone to pick it up whenever they walk past. The penalty timer is meant to have them deliver it instead. Leaving the packages at front ready for pickup also means a random passerby may tear them all open and lead to big penalties, further discouraging this method while giving cargo a reason to worry about protecting their mail.

> Additionally, mail is not meant to be a free item generator for cargo to use to flex drip they get. Losing money for opening mail that doesn't belong to them means they it is almost never worth it to open a package while not the recipient, other than possibly sabotaging cargo if they slip up.

Reviewer 3: ```suggestion
Additionally, mail is not meant to be a free item generator for cargo to use to flex drip they get. Losing money for opening mail that doesn't belong to them means that it is almost never worth it to open a package while not the recipient, other than possibly sabotaging cargo if they slip up.
```

- Encourage player interaction. Simple mail deliveries can greatly help with interactions between cargo and other players/departments.
- Add a little bit of chaos to the station. Mail ultimately can grant some silly random items to the recipient, and what is more fun than finding a bike horn or some bananas to annoy others with?

## Features to be added

### Mail Teleporter

The Mail Teleporter is the main thing controlling the flow of mail to the station. It should be mapped somewhere within cargo.
While connected to power and enabled, it will generate mail on top of itself every few minutes. Anyone with cargo access can toggle it on and off to prevent further mail from spawning.

Mail Teleporter is one of a kind machine, with only QM having a spare board to construct it, just like how cargo sell computers can be destroyed. This encourages actually taking care of it as to protect it from destruction by either having cargo keep a closer eye on it or putting it in a more secure area.

### Deliveries

Deliveries refer to anything that can spawn using the Mail Teleporter. The main focus here is Letters and Packages.

> On spawn, deliveries get assigned a random player in the crew manifest. When collected by the desired recipient they must scan their fingerprint on the delivery to open it, granting them a little reward and cargo some bonus money.

Reviewer 4: By finger print do you mean ID or just touching it or something? In delta its ID which has issues (Mail fraud with agent ID)

Author: Fingerprint as in you use it in-hand while you have no gloves on. If your fingerprint matches it unlocks and using it again will unwrap it.

Reviewer 5: Delta-V has you tap your ID to it. that way someone with your ID can steal your package. (also cryo players can "get" packages by just having someone steal their ID and openeing their packages)

Reviewer 2: A Fingerprint would work pretty well, plus with a Changeling it could be a funny way to out one who hadn't changed their fingerprints in a while. I do still think tying the reward to opening the letter itself maybe isn't the best route since a letter could be technically delivered but not give the reward if the person doesn't open it.

It wouldn't happen often but it's a frustration worth considering if someone's going to be a grump. Paradise had a Mail Scanner that let you scan a letter/package into a database to keep track of your current letters/packages, then scan the crewmember in question before handing it over to mark it as "delivered" and give Cargo the reward. Maybe a small fingerprint scanner?

If a non-recipient is to try to open a delivery, it will instead beep and say they are unable to open it. This however can be bypassed by tearing the delivery open (for letters) or smashing it open (for packages). Doing so will invoke a penalty on cargo, taking away from their funds, but will still grant the player whatever was inside.

#### Letters

A small item, usually containing spam mail, birthday wishes (maybe even with money!) and similiar.
They can be torn open using a Verb if the current holder is unable to open them. Obviously giving cargo a penalty (as stated above) but granting the user the contents.

#### Packages

> A big item that occupies both hands. They are supposed to contain things slightly more of value or items that can be useful to your average player. The loot table can include things like tools, plushies toys, food as well as some funny things like pipebombs or minor contraband. What needs to be kept in mind is that the "evil" or "funny" options need to be very unlikely as to not make people scared of opening packages, that would beat the entire point of the design.

Reviewer 2: Assuming packages would have a smaller spawn chance compared to letters but give a greater cash reward to cargo, correct?

Author: Yes thats correct

When on the ground it can be smashed open with any weapon to break the seal, allowing access to whatever is stored inside, still granting cargo a penalty.

### Future Expansions

Not meant to be full ready ideas or direction for the future, simply possible ways this can be improved to show mail has potential to be expanded on in the future.

#### Syndie Uplink Items

> Deliveries don't have to stop at just letters and packages. There is also the possibility of a cargo-only uplink item that is a rigged package, potentially spawning something hostile or exploding.

Reviewer 4: I would give a specific example of how you think this should work. E.g TC cost, and exactly how the abilities work. It can always be changed later, but I think giving a solid starting point is important

Reviewer 2: Agreed, since this could easily just be a regular package that the Syndie crew member puts something into and then sends to their intended target.

There could also be some way to sneak items into packages, having the recipient leave their sweet fingerprints on the contents and allowing for a way to frame them.

#### Custom Mail

> Ever wanted to give your arch-nemesis a threatening letter, but never wanted to give it in person? There is potential for a way to create custom letters/packages and allow cargo to deliver them for you.

Reviewer 4: Again, I would give a specific example of how this functions!

Reviewer 2: There's already a way to make letters in game and deliver them so this isn't particularly necessary, it's just very rarely used since there's not much direction for mail, nor is there a lot of dedicated mapped envelopes.

#### Modifiers

> Fragile packages that break when dropped? Letters that grow legs and beg you to not be delivered as that ends their life? Things like these can be added in the future for more variety.

Reviewer 4: Love this! Super fun idea. Still, giving like 5 very specific examples with full details would be good

Reviewer 2: Glitterbombs, letters that teleport you, mimic packages. bikeshedding but it's just fun ideas.

### Main things to consider:
- Delivery items need to have a timer that starts upon being taken out of the teleporter. When the timer runs out the money earned should be significatly lowered, either gradually or instantly. This encourages cargo to actually deliver the mail instead of leaving it at the front for others to check.
- Deliveries that are losing value due to not being delivered should grant less of a penalty to cargo when they do eventually get opened, someone stealing and keeping mail just to sabotage cargo when they find a moment to smash it open should not bring huge penalties.
- Items granted by deliveries cannot hold big value to them, as to not make breaking packages open and selling their contents more profitable than delivering them.

> - Cargo should have a reason to not allow their deliveries to be stolen, as that obviously not only makes them lose out of money, but also gives them a penalty if the seal is broken by a 3rd party.

Reviewer 2: Ideally the only penalty (since this is part of a low stress role) should be that there's no spesos given rather than Cargo being actively harmed. The act of delivering the mail already giving more Spesos than selling anything inside it is enough of an incentive for Cargo to want to get them delivered.

The profit motive is incentive enough for them to want to keep Mail flowing and unstolen, additional penalties arn't really necessary.

Author: I feel like this penalty is good so cargo doesn't simply leave their mail without a care, gives them a reason to worry about it being stolen and all that. The penalty doesn't have to be massive, maybe 250 spesos at best or something.

Reviewer 2: Yes but the issue is that cargo is already being punished for not delivering the mail by not being paid, further punishment is just going to introduce a level of friction that's unnecessary for a low stakes job, especially to a techie that might be still learning to navigate the station and will naturally be on the slower side. They already need to worry about it being stolen because it's a good profit margin. 

I'm all for good friction but losing money for not delivering the mail is just going to be an unnecessary point of frustration.

Reviewer 2: Especially when some letters might not be able to be delivered due to circumstances on station, or if the recipient dies after the letter is spawned (and updating the letter automatically when someone dies would be a bad idea on its own).

## Game Design Rationale

Currently cargo is oriented around the bounty system (except salvage, who is mostly separate from the main cargo loop). A mail system allows for more interaction between players as well as lowers dependency on bounties.

Cargo is already meant to interact with other departments, this plays into that fact and gives them a little something to do in the meantime. Ideal situation could be delivering a package to a botanist at the same time as you want to grab a bounty from them. Or a new player player grabbing a bunch of deliveries and getting to explore the station and meet other players as they deliver them.

## Roundflow & Player interaction

At the start of the round, a cargo technician visits Mail Teleporter and grabs letters into their bag and a few packages. Upon checking the manifest they may also pick some bounties to deliver to that person at the same time. The cargo tech brings the mail to the recipient, who then unlocks it and grants cargo some bonus money and gets something small in return.
In the case the delivery is simply never opened, it will slowly lose all value and become free game for whoever finds it and decides to open it.

> Interactions with other players are key here, both sides benefit while being able to get to interact more with each other. Cargo should not simply leave mail at their front door and get someone to pick it up whenever they walk past. The penalty timer is meant to have them deliver it instead. Leaving the packages at front ready for pickup also means a random passerby may tear them all open and lead to big penalties, further discouraging this method while giving cargo a reason to worry about protecting their mail.

Reviewer 2: It might be worth considering how the Mailing System would affect Cargo's mindset. If they have a few letters addressed to a certain department and it's a map with properly set up Mail bins, they might opt to just send mail that way along with a quick message to that department. The penalty system wouldn't be much of a discouragement since the speed of bulk delivery would outweigh that. (Personally since Mail systems are rarely mapped I'd consider this an A-Okay method for a busy cargo but it's worth noting.)

Author: I do not think sending packages via the mail chutes is a bad thing, but it also runs the risk that the incorrect person might get the package, or it will not be received at all. Ultimately delivery in person will always be the better and safer option.

> Additionally, mail is not meant to be a free item generator for cargo to use to flex drip they get. Losing money for opening mail that doesn't belong to them means that it is almost never worth it to open a package while not the recipient, other than possibly sabotaging cargo if they slip up.

Reviewer 2: I think the optics too of walking up on a cargo bay littered with opened envelopes and everyone playing with/wearing the equivalent of mcdonald's happy meal toys would be a pretty good discouragement method. 

Maybe instead of a direct penalty it keeps track of the amount of letters opened by Cargo quietly and then publicly shames them for disrespecting the sanctity of mail. Psychological discouragement is always a fun option.

Reviewer 3: remember kids, opening mail not addressed to you is a crime and you will be charged accordingly by the space postal service

Reviewer 5: ~~if cargo opens a ceartain ammount of mail the death squad is called on cargo in specific~~

# Cargo Mail Delivery

| Designers | Implemented | GitHub Links |
|---|---|---|
| ScarKy0 | :x: No | TBD |

## Overview

> This design doc is meant to introduce the concept of mail deliveries to the cargo gameplay loop. Allowing cargo to deliver mail to specific people on the station in order to gain some bonus money without the need to only run bounties.

Reviewer 3: Needs to reworded for clarity, it sounds like this used to be a single sentence

Author: me when having to write coherent sentences

When on the ground it can be smashed open with melee attacks to break the seal, allowing access to whatever is stored inside and obviously not granting cargo any money.

#### Fragile Deliveries

Fragile deliveries serve as a small modifier to add onto already spawned ones. A fragile delivery will break upon being thrown or hit, leading to it losing a major portion of it's value when delivered. Mail being fragile should be indicated on the sprite, as well as have a unique examine text.
The main point here is to add a little bit of variety to mail as to not keep it boring and too simple. While delivering items isn't the most exciting thing ever, it shouldn't be boring either.

#### Priority Mail

> Same as above, this serves as a small modifier for newly spawned mail. Instead of being broken on dropping, instead this delivery is on a timer.

Reviewer 3: Instead of instead this

# Cargo Mail Delivery

| Designers | Implemented | GitHub Links |
|---|---|---|
| ScarKy0 | :x: No | TBD |

## Overview

> The purpose of this design doc is to introduce the concept of mail deliveries to the cargo gameplay loop, allowing to earn bonus spesos by carrying around letters and packages and giving them to desired recipients.

Reviewer 3: I'm not sure if your wording here was intentionally specific to promote an actual giving action instead of enveloping the behavior around delivery. But you can have this suggestion as a treat.

```suggestion
The purpose of this design doc is to introduce the concept of mail deliveries to the cargo gameplay loop, allowing cargo to earn bonus spesos by delivering letters and packages to desired recipients.
```

Author: I kinda wanted to just avoid a "take a shot every time you hear 'delivery' " moment

Reviewer 3: don't ctrl + F for `flow` in the flowmos doc :godo:

This is by no means supposed to replace bounties, instead being an additional method of income as to not have cargo rely on them too much.

## Background

> Currently cargo's main way of earning money is the bounty system and selling random objects around the station. This can lead to situations such as other departments not cooperating or some bounties simply being not possible to make within a reasonable timeframe, as well as getting repetitive after a while. Times where most bounties are unachievable beyond simple ones (such as botany, but those take a while!) and the inability to skip due to a long cooldown happen often.

Reviewer 3: Reword in a bulleted list for clarity.
```suggestion
Currently, cargo's main way of earning money is the bounty system and selling random objects around the station. Unfortunately, this can lead to a lot of undesired gameplay:
- Departments can choose to not cooperate with fulfilling bounties, as they rarely get anything direct in return.
- Some bounties are impossible to procure in a reasonable timeframe.
- Most bounties are extremely repetitive and offer no more depth than the same repeat item requests.
Because of the reasons mentioned above, unachievable bounties happen often and bounty skips are used frequently.
```

While those issues can be fixed by making changes to the bounty system (like lowering the bounty skip cooldown) that feels like a bandaid fix to the fact cargo has basically only one way to earn money.

Delivering mail is meant to serve as a secondary option of income cargo can take care of during their shift.

> This is meant to serve several things:

Reviewer 3: ```suggestion
This new mechanic is intended to achieve several things:
```

- Add gameplay variety to cargo. Cargo as it stands has a rather simple gameplay loop. You print a bounty, you deliver the invoice to a department and then you sit at cargo until you're told it is ready for pickup. Not only is this repetitive but several bounties can also be completed on the spot, which makes cargo generally boring gameplay wise.
- Encourage player interaction. Simple mail deliveries can greatly help with interactions between cargo and other players/departments.
- Add a little bit of chaos to the station. Mail ultimately can grant some silly random items to the recipient, and what is more fun than finding a bike horn or some bananas to annoy others with?

## Features to be added

### Mail Teleporter

The Mail Teleporter is the main thing controlling the flow of mail to the station. It should be mapped somewhere within cargo.
While connected to power and enabled, it will generate mail on top of itself every few minutes. Anyone with cargo access can toggle it on and off to prevent further mail from spawning.

> The amount of deliveries spawned should depend on the amount of players on the station, with a minimum amount to ensure a steady flow of mail. Ideally moments where cargo is overflowing with mail, or getting too little to be able to make use of it, should never happen.

Reviewer 3: You should probably note some reasons why we explicitly don't want this to happen. Some good ones are:
- The mail department is intended to be a "side-hustle" and shouldn't overload a new player with gargantuan amounts of mail. It shouldn't be the main focus of the department, tending to department requests (material/item in/out) should.
- The crew continuously receiving mail makes "getting a package in the mail!" no longer exciting. It'll just become a normalized and "why should I care?" thing if you're receiving mail every 2 seconds.

Author: well its not a "mail department" but yeah ill include a lil reasoning

> Mail Teleporter is one of a kind machine, with only QM having a spare board to construct it, just like how cargo sell computers can be destroyed. This encourages actually taking care of it as to protect it from destruction by either having cargo keep a closer eye on it or putting it in a more secure area. In case an additional one is built, it should not generate additional mail, only split the existing amount between all available teleporters. This is to ensure cargo cannot simply create a mail farm.

Reviewer 3: ```suggestion
The Mail Teleporter is a one of a kind machine, with only QM having a spare board to construct it, just like how cargo sell computers can be destroyed. This encourages actually taking care of it as to protect it from destruction by either having cargo keep a closer eye on it or putting it in a more secure area. In case an additional one is built, it should not generate additional mail, only split the existing amount between all available teleporters. This is to ensure cargo cannot simply create a mail farm.
```

Reviewer 3: Just be aware that players pre-emptively making the mail teleporter secure because it's vital to the department shouldn't happen.

At shift start nobody should be moving it to the back of cargo. That's a weird thing to promote and it sets a standard that we don't want to follow IMO.

Author: I do not think this needs to be mentioned, you already have many one-of-a-kind computers and even so they are never really strictly reinforced or cared for much. However  I will remove the mention of putting it in more secure locations

### Deliveries

Deliveries refer to anything that can spawn using the Mail Teleporter. The main focus here is Letters and Packages.
On spawn, deliveries get assigned a random player in the crew manifest. When collected by the desired recipient they must scan their fingerprint on the delivery to open it, granting them a little reward and cargo some bonus money.

> If a non-recipient is to try to open a delivery, it will instead say they are unable to open it due to their fingerprint not matching. This however can be bypassed by tearing the delivery open (for letters) or smashing it open (for packages). Doing so will not grant cargo any points, as well as will damage anything inside of the delivery, but will still grant the unintended recipient the contents.

Reviewer 3: ```suggestion
If a non-recipient tries to open a delivery, it will instead say they are unable to open it due to their fingerprint not matching. This however can be bypassed by tearing the delivery open (for letters) or smashing it open (for packages). Doing so will not grant cargo any points, as well as will damage anything inside of the delivery, but it will still grant the unintended recipient the contents.
```

#### Letters

A small item, usually containing spam mail, birthday wishes (maybe even with money!) and similiar.
They can be torn open using a Verb if the current holder is unable to open them.
Letters are not meant to contain expensive items, usually sticking to some silly things (joints, spam, pills, etc.) or more rarely things of interest (research disks, pins, etc).

#### Packages

> A big item that occupies both hands. They are supposed to contain things slightly more of value or items that can be useful to your average player. The loot table can include things like tools, plushies toys, food as well as some funny things like pipebombs or rarely contraband. What needs to be kept in mind is that the "evil" or "funny" options need to be very unlikely as to not make people scared of opening packages, that would beat the entire point of the design. Last thing we want is someone being afraid of opening a package in case it explodes.

Reviewer 3: > A big item that occupies both hands.

Use an inventory size. How big should it be? Medium, Large, or Ginormous?

When on the ground it can be smashed open with melee attacks to break the seal, allowing access to whatever is stored inside and obviously not granting cargo any money.

#### Fragile Deliveries

Fragile deliveries serve as a small modifier to add onto already spawned ones. A fragile delivery will break upon being thrown or hit, leading to it losing a major portion of it's value when delivered. Mail being fragile should be indicated on the sprite, as well as have a unique examine text.
The main point here is to add a little bit of variety to mail as to not keep it boring and too simple. While delivering items isn't the most exciting thing ever, it shouldn't be boring either.

#### Priority Mail

Same as above, this serves as a small modifier for newly spawned mail. Instead of being broken on dropping, this delivery will start on a timer.
The exact time by which a piece of mail marked as priority needs to be delivered can be viewed on examine, showing time within the shift, rather than a countdown. If that time is passed, the mail will lose value and earn cargo less.
Obviously, this should also be indicated by a change on the sprite.

### Future Expansions

> Not meant to be full ready ideas or direction for the future, simply possible ways this can be improved to show mail has potential to be expanded on in the future.

Reviewer 3: Maybe just put one or two potential points about science research oriented for the mail subdepartment.

This is intended for contributors who are looking through these merged ddocs and looking for guidance on what they could implement that benefits mail.

Author: Fuck science.

(To be fair, I don't know what could be researchable for mail. I know downstream they have the mail clakes but I'm not a big fan of the idea)

#### More Modifiers

Fragile and Priority mail shouldn't be the only things that can affect deliveries. If more variety is needed, this system can be made more exciting by adding possible new modifiers. Letters that grow legs and beg you to not be delivered as that ends their life? Things like these can be added in the future.
A good example of such implementation could be that whenever a package there is a low chance it will receive a modifier. Such modifier will add onto the value, make the item pool more interesting but also make the package itself more annoying to handle. The following are only possible examples.
- Burning: The package releases intense heat into the atmosphere around it, but grants 2x spesos.
- Everlasting Sorrow: The package cries when carried, causing the carrier to slip on water if they're not careful for 1.25x spesos. Would mix great with being fragile.
- Glued: It cannot be dropped by itself without destroying it and must instead be "stripped" off the carrier for 1.25x spesos.
- Drunk: Constantly injects small amounts of ethanol into whoever holds it, making them drunk while in inventory for 1.5x spesos.

## Game Design Rationale

> Currently cargo is oriented around the bounty system (except salvage, who is mostly separate from the main cargo loop). A mail system allows for more interaction between players as well as lowers dependency on bounties, while also allowing new players to learn basics of the game.

Reviewer 3: ```suggestion
Currently cargo is oriented around the bounty system (except salvage, who is mostly separate from the main cargo loop). A mail system allows for more interaction between players as well as lowers dependency on bounties, while also allowing new players to learn the basics of the game.
```

> Cargo is already meant to interact with other departments, this plays into that fact and gives them a little something to do in the meantime. Ideal situation could be delivering a package to a botanist at the same time as you want to grab a bounty from them. Or a new player player grabbing a bunch of deliveries and getting to explore the station and meet other players as they deliver them.

Reviewer 3: ```suggestion
Cargo is already meant to interact with other departments, this plays into that fact and gives them a little something to do in the meantime. An ideal situation could be delivering a package to a botanist at the same time as you want to grab a bounty from them. Or a new player player grabbing a bunch of deliveries and getting to explore the station and meet other players as they deliver them.
```

## Roundflow & Player interaction

> At the start of the round, after getting prepared a cargo technician visits Mail Teleporter and grabs letters into their bag and a package. Upon checking the manifest they may also pick some bounties to deliver at the same time. The cargo tech brings the mail to the recipient, who then unlocks it and grants cargo some bonus money and gets something small in return.

Reviewer 3: ```suggestion
At the start of the round after getting prepared, a cargo technician visits the Mail Teleporter and grabs some letters, stuffs them into their bag, and picks up a package. Upon checking the manifest they may also pick up some bounties to deliver at the same time. The cargo tech brings the mail to the recipient, who then unlocks it and grants cargo some bonus money and gets something small in return. The cargo tech also brings up the bounty they previously selected, offering a follow-up to the department later in the shift. Future mail deliveries to the department encourage cargo to follow up with bounties previously assigned to departments.
```

## Overview

The purpose of this design doc is to introduce the concept of mail deliveries to the cargo gameplay loop, allowing cargo to earn bonus spesos by delivering letters and packages to desired recipients.
This is by no means supposed to replace bounties, instead being an additional method of income as to not have cargo rely on them too much.

## Background

Currently, cargo's main way of earning money is the bounty system and selling random objects around the station. Unfortunately, this can lead to a lot of undesired gameplay:
- Departments can choose to not cooperate with fulfilling bounties, as they rarely get anything direct in return.
- Some bounties are impossible to procure in a reasonable timeframe, given current circumstances.
- Most bounties are repetitive after a while and offer no more depth than the same repeat item requests.

Because of the reasons mentioned above, unachievable bounties happen often and bounty skips are used frequently.

Delivering mail is meant to serve as a secondary option of income cargo can take care of during their shift.

> This new mechanic to inteded to achieve several things:

Reviewer 3: ```suggestion
This new mechanic is intended to achieve several things:
```

### Mail Teleporter

The Mail Teleporter is the main thing controlling the flow of mail to the station. It should be mapped somewhere within cargo.
While connected to power and enabled, it will generate mail on top of itself every few minutes. Anyone can toggle it on and off to prevent further mail from spawning.

> The amount of deliveries spawned should depend on the amount of players on the station, with a minimum amount to ensure a steady flow of mail. Cargo should never have so much mail to the point it becomes overwhelming, it is not supposed to be the main point of the department, and should never have little enough of it that it brings nothing new to the cargo experience.

Reviewer 3: ```suggestion
The amount of deliveries spawned should depend on the amount of players on the station, with a minimum amount to ensure a steady flow of mail. Cargo should never have so much mail to the point it becomes overwhelming, it is not supposed to be the main point of the department, and it should never have little enough of it that it brings nothing new to the cargo experience.
```

> The Mail Teleporter is a one of a kind machine, with only QM having a spare board to construct it, just like how cargo sell computers can be destroyed. This encourages actually taking care of it as to protect it from destruction by having cargo keep a closer eye on it, as to not lose another source of income. In case an additional one is built, it should not generate additional mail, only split the existing amount between all available teleporters. This is to ensure cargo cannot simply create a mail farm as well as to ensure deliveries stay fresh and don't overwhelm the crew. In cases where too much mail keeps being delivered to someone, they might start ignoring it as it becomes an annoyance.

Reviewer 3: that one sentence was hard to follow

```suggestion
The Mail Teleporter is a one of a kind machine, with only the QM having a spare board to construct it, just like how cargo sell computers can be destroyed. This encourages actually taking care of it due to it being a second source of passive income, and promotes frequent visits to the Mail Teleporter. In case an additional one is built, it should not generate additional mail, only split the existing amount between all available teleporters. This is to ensure cargo cannot simply create a mail farm as well as to ensure deliveries stay fresh and don't overwhelm the crew. In cases where too much mail keeps being delivered to someone, they might start ignoring it as it becomes an annoyance.
```
